### PR TITLE
Add support for colliding with the ground in the C++ code

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: ksp_plugin/part_subsets.hpp [hunk 1]
```diff
@@ -43,6 +43,13 @@ class Subset<ksp_plugin::Part>::Properties final {
   // Maintains |parts_| by joining the lists.
   void MergeWith(Properties& other);
 
+  // "What’s this thing suddenly coming towards me very fast? Very very fast.
+  // So big and flat and round, it needs a big wide sounding name like … ow …
+  // ound … round … ground! That’s it! That’s a good name – ground!  I wonder if
+  // it will be friends with me?"
```
Comment (eggrobin) [inline ksp_plugin/part_subsets.hpp]: Enclose in  `“”` rather than `""`.

File: ksp_plugin/plugin.hpp [hunk 2]
```diff
@@ -203,12 +203,15 @@ class Plugin {
   // Calls |MakeSingleton| for all parts in loaded vessels, enabling the use of
   // union-find for pile up construction.  This must be called after the calls
   // to |IncrementPartIntrinsicForce|, and before the calls to
-  // |ReportCollision|.
+  // |ReportPartCollision|.
```
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: `or |ReportGroundCollision|`

File: ksp_plugin/plugin.hpp [hunk 3]
```diff
@@ -203,12 +203,15 @@ class Plugin {
...
-  // Notifies |this| that the given vessels are touching, and should gravitate
+  // Notifies |this| that the given part is touching the ground.
+  virtual void ReportGroundCollision(PartId part) const;
+
+  // Notifies |this| that the given parts are touching, and should gravitate
   // as part of a single rigid body.
-  virtual void ReportCollision(PartId part1, PartId part2) const;
+  virtual void ReportPartCollision(PartId part1, PartId part2) const;
 
   // Destroys the vessels for which |InsertOrKeepVessel| has not been called
```
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: Also the vessels which transitively touch the ground.

File: ksp_plugin/part_subsets.hpp (hunk 1 above)
Comment (pleroy) [reply ksp_plugin/part_subsets.hpp]: Done.

File: ksp_plugin/plugin.hpp (hunk 2 above)
Comment (pleroy) [reply ksp_plugin/plugin.hpp]: Done.

File: ksp_plugin/plugin.hpp (hunk 3 above)
Comment (pleroy) [reply ksp_plugin/plugin.hpp]: Done.

**Pull request merged**